nixos/modules/services/monitoring/prometheus/exporters/sabnzbd.nix

```diff
@@ -30,18 +34,24 @@ in
   serviceOpts =
     let
       servers = lib.zipAttrs cfg.servers;
-      apiKeys = lib.concatStringsSep "," (builtins.map (file: "$(cat ${file})") servers.apiKeyFile);
+      credentials = lib.imap0 (i: v: { name = "apikey-${toString i}"; path = v; }) servers.apiKeyFile;
     in
     {
+      serviceConfig.LoadCredential = builtins.map ({ name, path }: "${name}:${path}") credentials;
+
       environment = {
         METRICS_PORT = toString cfg.port;
         METRICS_ADDR = cfg.listenAddress;
         SABNZBD_BASEURLS = lib.concatStringsSep "," servers.baseUrl;
       };
 
-      script = ''
-        export SABNZBD_APIKEYS="${apiKeys}"
-        exec ${lib.getExe pkgs.prometheus-sabnzbd-exporter}
-      '';
+      script =
+        let
+          apiKeys = lib.concatStringsSep "," (builtins.map (cred: "$(cat $CREDENTIALS_DIRECTORY/${cred.name})") credentials);
+        in
+        ''
+          export SABNZBD_APIKEYS="${apiKeys}"
```

Comment:
In case someone else is wondering as well:
For a moment I thought that the secrets need to be passed via environment variable in an unordered list, because it looked like there is no relation between the API keys and the server URLs. I assumed that it would be preferable to use a single user-provided `EnvironmentFile` (**systemd.exec(5)**) instead.
But after looking at [the exporter's docs](https://github.com/msroest/sabnzbd_exporter/blob/0.1.70/README.md#runningdeveloping), I saw that API keys and URLs are matched by their respective position in the env vars.

The only remaining thing I see would be to replace `$(cat /path/to/file)` with `$(< /path/to/file)` in the script, letting the shell do the reading instead of forking to coreutils like:
`apiKeys = lib.concatMapStringsSep "," (cred: "$(<$CREDENTIALS_DIRECTORY/${cred.name}") credentials`

Reply:
Thank you for the review, I made the change and rebased.